src/ln/channelmonitor.rs

```diff
@@ -119,27 +125,70 @@ pub struct SimpleManyChannelMonitor<Key> {
 	chain_monitor: Arc<ChainWatchInterface>,
 	broadcaster: Arc<BroadcasterInterface>,
 	pending_events: Mutex<Vec<events::Event>>,
+	htlc_sources: Mutex<HashMap<[u8; 32], (Key, HTLCPreviousHopData)>>,
```

Comment:
Hmmm, htlc_sources can grow unbounded, no? I dont think we need to track this outside of the ChannelMonitor itself or for historical transactions, though. Any attempt by our counterparty to broadcast a revoked state we can just treat as all-HTLCs-failed and fail backwards whatever HTLCs we have pending, and then we can just keep a set of HTLCPreviousHopData in each ChannelMonitor itself for the latest local and latest remote transactions.

Reply:
Ah dumb me it grow unbounded, not all tx are gonna to settle onchain obviously, should be doable to only keep last for latest {local, remote} commitment tx in ChannelMonitor. Just wait to be fixed on other issue on funding_txo, may change due to that.